filter.go

```diff
@@ -66,3 +76,112 @@ func (cf CorrelationFilter) ToFilterDescription() FilterDescription {
 		CorrelationFilter: cf,
 	}
 }
+
+// MarshalXML implements the xml.Marshaller interface for CorrelationFilterProperties.
+func (c CorrelationFilterProperties) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
```

Comment:
Looking closer at the C# impl, I think this needs a bit more work.

Reply:
What's missing compared to the C# impl?

Reply:
There's handling of XML namespace stuffs, see [here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Microsoft.Azure.ServiceBus/src/Filters/XmlObjectConvertor.cs#L99).